ui/lib/kv/addon/routes/secret/metadata.js

```diff
@@ -7,31 +7,67 @@ import Route from '@ember/routing/route';
 import { service } from '@ember/service';
 
 export default class KvSecretMetadataRoute extends Route {
-  @service store;
+  @service capabilities;
   @service secretMountPath;
+  @service store;
 
   fetchMetadata(backend, path) {
     return this.store.queryRecord('kv/metadata', { backend, path }).catch((error) => {
       if (error.message === 'Control Group encountered') {
         throw error;
       }
-      return {};
+      return null;
     });
   }
 
+  async fetchCapabilities(backend, path) {
+    const metadataPath = `${backend}/metadata/${path}`;
+    const dataPath = `${backend}/data/${path}`;
+    const resp = await this.capabilities.fetchMultiplePaths([metadataPath, dataPath]);
```

Comment:
Thinking about this more, maybe `fetchMultiplePaths` could return an object keyed by the paths, with the model as the value? just spitballing on the new pattern

Reply:
Yeah, that's a good idea. I was also wondering about just returning the permissions instead of the full model 🤔 If someone needs the list of capabilities, they can use the base `request` method